tools/autograd/gen_autograd_functions.py

```diff
@@ -82,6 +77,150 @@
 }
 """)
 
+# Generates python bindings
+#
+# This generates the definitions for:
+#   (1) The PyTypeObject for each backward grad_fn subclassing Node
+#   (2) The entry for PyTypeObject's tp_getset slot (an array of PyGetSetDef structs)
+#       We generate one PyGetSetDef struct for each of grad_fn's saved inputs and outputs
+#       Each PyGetSetDef has a function ptr to a getter, also defined here (3).
+#   (3) Getters for each of grad_fn's saved inputs and outputs.
+#
+PY_FUNCTION_DEFINITION = CodeTemplate("""\
+static PyTypeObject ${op}Class;
+addClass<${op}>(${op}Class, "${op}", ${op}_properties);
+""")
+
+PY_FUNCTION_PROPS_AND_GETTERS = CodeTemplate("""\
+${all_getter_definitions}
+
+static struct PyGetSetDef ${op}_properties[] = {
+  THP_FUNCTION_DEFAULT_PROPERTIES,
+  ${all_getsetdef_structs}
+  {nullptr} /* sentinel */
+};
+
+""")
+
+PY_GETSETDEF_STRUCT = CodeTemplate("""\
+{(char*)"_${name}", (getter)THP${op}_${name}_getter, nullptr, nullptr, nullptr}""")
```

Comment:
Since one of the goal here is to allow the users to easily find these irrespective of their name, maybe we should consider giving them a common prefix like: `_saved_foo`. This will allow users to use `dir(Node)` and filter anything that starts with `_saved_` to get all the attributes they want.
What do you think?